crates/cext/src/circuit.rs

```diff
@@ -471,6 +512,90 @@ pub unsafe extern "C" fn qk_circuit_gate(
     ExitCode::Success
 }
 
+/// @ingroup QkCircuit
+/// Append a ``QkGate`` with ``QkParam*`` parameters to the circuit.
+///
+/// @param circuit A pointer to the circuit to add the gate to.
+/// @param gate The ``QkGate`` to add to the circuit.
+/// @param qubits The pointer to the array of ``uint32_t`` qubit indices to add the gate on. This
+/// can be a null pointer if there are no qubits for ``gate`` (e.g. ``QkGate_GlobalPhase``).
+/// @param params The pointer to the array of ``QkParam*`` parameters to use for the gate parameters.
+/// This can be a null pointer if there are no parameters for ``gate`` (e.g. ``QkGate_H``).
+///
+/// @return An exit code.
+///
+/// # Example
+///
+/// ```c
+/// QkCircuit *qc = qk_circuit_new(100, 0);
+/// QkParam *theta = qk_param_new_symbol("theta");
+/// uint32_t qubit[1] = {0};
+/// const QkParam* params[1] = {theta};
+/// qk_circuit_gate_param(qc, QkGate_RX, qubit, params); // add RX(theta) to the circuit
+/// ```
+///
+/// # Safety
+///
+/// The ``qubits`` and ``params`` types are expected to be a pointer to an array of ``uint32_t``
+/// and ``QkParam*`` respectively where the length is matching the expectations for the standard
+/// gate. If the array is insufficiently long the behavior of this function is undefined as this
+/// will read outside the bounds of the array. It can be a null pointer if there are no qubits
+/// or params for a given gate. You can check ``qk_gate_num_qubits`` and ``qk_gate_num_params`` to
+/// determine how many qubits and params are required for a given gate.
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit``,
+/// or if any of the elements in the ``params`` array is not a valid, non-null pointer to a
+/// ``QkParam``.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_circuit_gate_param(
```

Comment:
Mentioned offline, but I'd prefer something like `qk_circuit_generic_gate` or `qk_circuit_parameterized_gate`, the important part being that we end with a noun of what is being added, since that's what we're doing for everything else in `QkCircuit`. We aren't adding a "gate param" to the circuit, we're adding a "parameterized gate".